subgraphs/uniswap-forks/protocols/spookyswap/src/common/handlers/handleRewardV2.ts

```diff
@@ -12,20 +12,52 @@ import {
   convertTokenToDecimal,
   roundToWholeNumber,
 } from "../../../../../src/common/utils/utils";
-import { getOrCreateMasterChef } from "../../../../../src/common/masterchef/helpers";
+import {
+  getOrCreateMasterChef,
+  getOrCreateMasterChefStakingPool,
+} from "../../../../../src/common/masterchef/helpers";
 
 // Updated Liquidity pool staked amount and emmissions on a deposit to the masterchef contract.
 export function updateMasterChef(
   event: ethereum.Event,
   pid: BigInt,
   amount: BigInt
 ): void {
-  const masterChefV2Pool = _MasterChefStakingPool.load(
-    MasterChef.MASTERCHEFV2 + "-" + pid.toString()
-  )!;
+  const poolContract = MasterChefV2Spookyswap.bind(event.address);
+  const masterChefV2Pool = getOrCreateMasterChefStakingPool(
+    event,
+    MasterChef.MASTERCHEFV2,
+    pid
+  );
+
   const masterchefV2Contract = MasterChefV2Spookyswap.bind(event.address);
   const masterChefV2 = getOrCreateMasterChef(event, MasterChef.MASTERCHEFV2);
 
+  // Sometimes the pool addition event is not emitted before the deposit/withdraw event. In this case, we need to add the pool and allocation to the masterchef entity.
+  if (!masterChefV2Pool.poolAddress) {
+    const poolAddress = poolContract.try_lpToken(pid);
+    const poolInfo = poolContract.try_poolInfo(pid);
+    if (!poolAddress.reverted) {
+      masterChefV2Pool.poolAddress = poolAddress.value.toHexString();
+    }
+    if (!poolInfo.reverted) {
+      masterChefV2Pool.poolAllocPoint = poolInfo.value.getAllocPoint();
+      masterChefV2.totalAllocPoint = masterChefV2.totalAllocPoint.plus(
+        poolInfo.value.getAllocPoint()
+      );
+    }
+    masterChefV2Pool.save();
+    masterChefV2.save();
+
+    if (!masterChefV2Pool.poolAddress) {
+      log.warning(
+        "poolInfo reverted: Could not find pool address for masterchef pool",
+        []
+      );
+      return;
+    }
```

Comment:
NBD, but I'd suggest moving this whole block into its own function

Reply:
Sure - I added a function for it.